### DISCUSSION TITLE
Source.getPlaylistTracks выдает 50 лишних треков

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#94] [Помощь с ошибкой] [answered]
Заметил это у себя, наверно, в течение недели. Может, проглядел и раньше началось.

`Source.getPlaylistTracks` выдает ровно на 50 треков больше, чем есть в плейлисте (добавляет дублей).
`let tracks= Source.getPlaylistTracks('name', 'id');`
Не воспроизвелось на 30, 50, 100 треках в плейлисте. Воспроизвелось на 365, 477, 860 треках в плейлисте.

У кого-то есть еще такая ошибка или я что-то намудрил у себя вновь?

**Comment by Chimildic** [comment, marked as answer]
Сделал [правку](https://github.com/Chimildic/goofy/commit/23353f5a722cf96f66c0d225724e26d415c9a243). Должно помочь.

**Reply by efremych** [reply]
Спасибо! Наладилось.

**Reply by Chimildic** [reply]
[Изменил](https://github.com/Chimildic/goofy/commit/28085e8b3b8e15fb5fdca47e86a7e9dc93904c02) еще раз, чтобы лишний запрос не отправлялся.

**Reply by ilyanikolaich** [reply]
Исправление подтверждаю.
Сегодня днем поймал глюк - думал крыша едет. Читаю плейлист, количество треков - 330. В самом плейлисте - 280. Ерунда какая-то.. Делаю дедупликейт треков - остается 280. Сохраняю как новый. В сохраненном - 280 треков. Хорошо. А теперь, дай-ка, думаю, перепроверю - читаю треки уже из нового плейлиста - их 330. Дедупликейт, запись еще одного нового плейлиста - 280 треков. Перепроверять побоялся.
После прочитанного выше - обновил код, проверил - в считываемом плейлисте - 280 треков. Выдохнул. Хорошо-то как. Большая благодарность нашему Chimildic-у!

**Reply by efremych** [reply]
да, а у меня описание плейлиста меняется, если количество треков в плейлисте не совпадает с количеством треков из описания. Ничего не добавлял/не удалял из плейлиста, а смотрю, описание всё обновляется и обновляется изо дня в день. Пошёл искать почему, и вот...
